### PR TITLE
Add --accumulate-attribute to tippecanoe-overzoom

### PR DESCRIPTION
**Pull request by e-n-f**
This allows, for instance, cluster size summation for heatmaps to work when overzooming tiles that were generated with a retain points multiplier.

In addition, when tiles are generated with a multiplier and a feature order, the multiplier clusters are still sorted into that order, but the features within the cluster remain in spatial sequence instead, so that the default feature for each multiplier cluster remains the same feature that would have been included in the tile if there were no multiplier.

### CODE REVIEW

File: attribute.cpp [hunk 1]
```diff
@@ -0,0 +1,146 @@
+#include <string>
+#include <map>
+#include "attribute.hpp"
+#include "errors.hpp"
+#include "serial.hpp"
+#include "jsonpull/jsonpull.h"
+#include "milo/dtoa_milo.h"
+
+void set_attribute_accum(std::map<std::string, attribute_op> &attribute_accum, std::string name, std::string type) {
```
Comment (e-n-f) [inline attribute.cpp:9]: previously in main.cpp

File: attribute.cpp [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+		t = op_comma;
+	} else {
+		fprintf(stderr, "Attribute method (%s) must be sum, product, mean, max, min, concat, or comma\n", type.c_str());
+		exit(EXIT_ARGS);
+	}
+
+	attribute_accum.insert(std::pair<std::string, attribute_op>(name, t));
+}
+
+void set_attribute_accum(std::map<std::string, attribute_op> &attribute_accum, const char *arg, char **argv) {
```
Comment (e-n-f) [inline attribute.cpp:34]: previously in main.cpp

File: attribute.cpp [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+		exit(EXIT_ARGS);
+	}
+
+	std::string name = std::string(arg, s - arg);
+	std::string type = std::string(s + 1);
+
+	set_attribute_accum(attribute_accum, name, type);
+}
+
+void preserve_attribute(attribute_op op, std::string &key, serial_val &val, std::vector<std::string> &full_keys, std::vector<serial_val> &full_values, std::map<std::string, accum_state> &attribute_accum_state) {
```
Comment (e-n-f) [inline attribute.cpp:82]: previously in tile.cpp

**Review by ChrisLoer — APPROVED** [review]
Erica gave me a tour of the code and I tested in Felt using an Earthquakes upload to see that with this change tippecanoe-overzoom could take tiles using retain-points-multiplier and make tiles whose cluster sizes were identical to tiles made without retain-points-multiplier.

**Pull request merged**